scripts/linters/pylint_extensions.py

```diff
@@ -2246,6 +2246,36 @@ def visit_classdef(self, node):
                 node=node, args=(node.name))
 
 
+class DisallowedImportsChecker(checkers.BaseChecker):
+    """Check that disallowed imports are not made."""
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'disallowed-imports'
+    priority = -1
+    msgs = {
+        'C0039': (
+            'Please use str instead of typing.Text',
+            'disallowed-text-import',
+            'Disallow import of Text from typing module',
+        ),
+    }
+
+    def visit_importfrom(self, node):
+        """Visits all import-from statements in a python file and ensures that
+        only allowed imports are made.
+
+        Args:
+            node: astroid.node_classes.ImportFrom. Node for a import-from
+                statement in the AST.
+        """
+        if node.modname != 'typing':
+            return
+        for (name, _) in node.names:
+            if name == 'Text':
+                self.add_message('disallowed-text-import', node=node)
```

Comment:
Should we also verify that someone doesn't try to use `import typing; typing.Text`?

Reply:
Do you think this is common? Haven't seen anyone do like this.

Reply:
@vojtechjelinek Should I change the error message to `Please use str instead of Text`? If we remove all the occurrences of Text, people won't use it mostly. For completeness it seems logical to disallow usage of typing.Text, but it will come at a cost - it requires more complexity (hence increased runtime) on linter side since now the scope of typing.Text could be anywhere in the codebase. Let me know what you think.

Reply:
Hmm, yeah I just realized that we would need to parse whole files instead of just imports. Let's keep it simple. I agree with changing the error message.

Reply:
Changed the message. Also please keep this thread unresolved for future visibility.